segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+from pathlib import Path
+import random
+from typing import Union, Tuple
+import os.path
+import math
+import copy
+
+import h5py
+import cv2
+import numpy as np
+import imageio_ffmpeg as mpg
+
+
+def downsample_h5_video(video_path: Union[str, Path], input_fps: int,
+                     output_fps: int, strategy: str = 'random') -> np.ndarray:
+    """
+    Function to down-sample an array stored in h5 format to a desired FPS.
+    Follows the strategy specified in the function call, current strategies are
+    (random, maximum, average). Random takes a random sample of one frame from
+    window, maximum takes the maximum frame in window, average takes the mean
+    of window.
+    Args:
+        video_path: The path to the input video in str or Path format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if os.path.exists(video_path):
+        path_parts = str(video_path).split('.')
+        extension = path_parts[len(path_parts) - 1]
+        if extension == 'h5':
+            with h5py.File(video_path, 'r') as open_video:
+                video = open_video['data'][:]
+                return _down_sample_array(full_array=video,
+                                          input_fps=input_fps,
+                                          output_fps=output_fps,
+                                          strategy=strategy)
+        else:
```

Comment:
maybe to-taste, but I don't really like burying function returns inside conditionals. Your conditionals could raise exceptions earlier and the function could end with a return.

Reply:
I agree, changed the structure, now raises exceptions first then proceeds if not exceptions